input/drivers/udev_input.c

```diff
@@ -505,38 +505,39 @@ static bool udev_input_add_device(udev_input_t *udev,
 
    strlcpy(device->devnode, devnode, sizeof(device->devnode));
 
-   /* Touchpads report in absolute coords. */
-   if (type == UDEV_INPUT_TOUCHPAD)
-   {
-      if (ioctl(fd, EVIOCGABS(ABS_X), &absinfo) < 0 ||
-            absinfo.minimum >= absinfo.maximum)
-         goto error;
-
-      device->mouse.x_min = absinfo.minimum;
-      device->mouse.x_max = absinfo.maximum;
-
-      if (ioctl(fd, EVIOCGABS(ABS_Y), &absinfo) < 0 ||
-          absinfo.minimum >= absinfo.maximum)
-         goto error;
-
-      device->mouse.y_min = absinfo.minimum;
-      device->mouse.y_max = absinfo.maximum;
-   }
+  
    /* UDEV_INPUT_MOUSE may report in absolute coords too */
-   else if (type == UDEV_INPUT_MOUSE && ioctl(fd, EVIOCGABS(ABS_X), &absinfo) >= 0)
+   if (type == UDEV_INPUT_MOUSE || type == UDEV_INPUT_TOUCHPAD )
    {
-      if (absinfo.minimum >= absinfo.maximum)
-         goto error;
-
-      device->mouse.x_min = absinfo.minimum;
-      device->mouse.x_max = absinfo.maximum;
-
-      if (ioctl(fd, EVIOCGABS(ABS_Y), &absinfo) < 0 ||
-          absinfo.minimum >= absinfo.maximum)
-         goto error;
+      if (ioctl(fd, EVIOCGABS(ABS_X), &absinfo) >= 0)
+      {      
+	
+        if (absinfo.minimum >= absinfo.maximum )
+      	{
+          device->mouse.x_min = 0;
+          device->mouse.x_max = 0;
+        }
+	else
+	{
+	  device->mouse.x_min = absinfo.minimum;
+	  device->mouse.x_min = absinfo.maximum; 
+        }         
+      }
 
-      device->mouse.y_min = absinfo.minimum;
-      device->mouse.y_max = absinfo.maximum;
+    if (ioctl(fd, EVIOCGABS(ABS_Y), &absinfo) >= 0)
+      {      
+	
+        if (absinfo.minimum >= absinfo.maximum )
+      	{
+          device->mouse.y_min = 0;
+          device->mouse.y_max = 0;
+        }
+	else
```

Comment:
Formatting is wrong. Use 3 spaces for intendation.

Reply:
i didnt realllise we had a format here good to know